File: BinaryTrees/CountSubtreesWithSum/count.py
class Solution:
    def __init__(self):
        self.ans = 0

    def giveSumNodeValue(self, node, target):
        if node is None:
            return 0

        lSum = self.giveSumNodeValue(node.left, target)
        rSum = self.giveSumNodeValue(node.right, target)

        total = lSum + rSum + node.data

        if total == target:
            self.ans += 1

        return total

    def countSubtrees(self, root, x):
        self.giveSumNodeValue(root, x)
        return self.ans

File: BinaryTrees/CountSubtreesWithSum/count.cpp
class Node {
  public:
    int data;
    Node* left;
    Node* right;
    Node(int val) {
        data = val;
        left = right = nullptr;
    }
};

class Solution {
  public:
    int ans = 0;
    int giveSumNodeValue(Node* node, int target){
        if(!node)return 0;
        int lSum = giveSumNodeValue(node->left,target);
        int rSum = giveSumNodeValue(node->right,target);
        int sum = lSum + rSum + node->data;
        if(sum == target)ans++;
        return sum;
    }

    int countSubtrees(Node* root, int x) {
        // code here
        giveSumNodeValue(root, x);
        return ans;
    }
};